radixdlt/src/main/java/com/radixdlt/network/hostip/HostIp.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.network.hostip;
+
+import java.util.Optional;
+
+/**
+ * Interface for obtaining a hosts IP.
+ */
+public interface HostIp {
+	/**
+	 * Return a host IP, if possible.
+	 * The underlying query mechanism is used to determine a IP address for
+	 * the host.
+	 * @return An optional host IP address, if one can be determined
+	 */
+	Optional<String> hostIp();
```

Comment:
Given that this could require a network connection, perhaps this can use a `Single` or `Maybe` object instead of being a blocking api. But not necessary for now.

Reply:
Given that it happens at setup time for now, I think blocking is fine.

Later it would be an improvement if an observable of IPs was provided so that things like dynamic IPs could be handled more nicely.